test/test_dynamic_transform.cpp

```diff
@@ -618,4 +618,217 @@ TEST_F(NVFuserTest, DynamicTransform11_CUDA) {
+
+  // Return pair of: number of concretizations & total number of kernel runtimes
+  auto countConcretizations = [&fusion_executor_cache]() {
+    std::unordered_set<const std::pair<
+        size_t,
+        std::optional<DynamicTransformConcretizationInfo>>*>
+        concs;
+    for (auto& it : fusion_executor_cache.getKernelRuntimes()) {
+      concs.insert(&it.first);
+    }
+    return concs.size();
+  };
+  size_t num_concretizations = countConcretizations();
+  // Check that concretizations and runtimes are cache misses only when they
+  // should be
+  auto checkCache = [&countConcretizations,
+                     &num_concretizations](bool expect_miss) {
+    auto current = countConcretizations();
+    ASSERT_EQ(current, num_concretizations + (size_t)expect_miss);
+    num_concretizations = current;
+  };
+
+  for (auto& inv : invocations) {
+    auto input_shape = std::get<0>(inv);
+    auto output_shape = std::get<1>(inv);
+    auto expect_miss = std::get<2>(inv);
+
+    auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+    auto bias_shape = (reshape_before_reduction) ? input_shape : output_shape;
+    at::Tensor at_x = at::randn(input_shape, options);
+    at::Tensor at_bias = at::randn(bias_shape, options);
+    std::vector<c10::IValue> aten_inputs = {at_x, at_bias};
+    // Add input scalars describing the reshape size for concretization
+    for (int i : c10::irange(output_dims)) {
+      aten_inputs.push_back(output_shape[i]);
+    }
+
+    auto outputs = fusion_executor_cache.runFusionWithInputs(aten_inputs);
+    checkCache(expect_miss);
+
+    auto at_tv1 = (reshape_before_reduction) ? (at_x + at_bias)
+                                             : at::sum(at_x, kReductionAxis);
+    auto at_x_reshape = at::native::view(at_tv1, output_shape);
+    auto at_y = (reshape_before_reduction)
+        ? at::sum(at_x_reshape, kReductionAxis)
+        : at::add(at_x_reshape, at_bias);
+
+    testValidate(&fusion, outputs, aten_inputs, {at_y}, __LINE__, __FILE__);
+  }
+}
+
+TEST_F(NVFuserTest, FusionDynamicReshapeReductionShmoo_CUDA) {
+  std::vector<std::pair<std::vector<int64_t>, std::vector<int64_t>>>
+      all_dynamic_reshape_examples = {};
+  for (auto invocations : std::vector<std::vector<dynamic_view_invocation>>{{
+           {{8, 3 * 4, 7, 9}, {8, 3 * 4, 7, 9}, true}, // trivial
+           {{8, 3 * 4, 7, 5}, {8, 3 * 4, 7, 5}, false}, // trivial
+           {{8, 3 * 4, 7, 9}, {8, 3, 4, 7 * 9}, true}, // merge(2) osplit(1, 3)
+           {{8, 3 * 4, 7, 9},
+            {8, 3, 4 * 7, 9},
+            true}, // merge(1) merge(2) osplit(1, 3)
+           {{8, 3 * 4, 7, 5},
+            {8, 3, 4 * 7, 5},
+            false}, // merge(1) merge(2) osplit(1, 3)
+           {{8, 3 * 5, 7, 9}, {8, 3, 5 * 7, 9}, false}, // merge(1) osplit(1, 3)
+           // test passing -1 dynamically for dimension size
+           //{{8, 3 * 5, 7, 9}, {8, 3, -1, 9}, false} // merge(1) osplit(1, 3)
```

Comment:
@naoyam I have commented this case. I am unsure if we want to support `-1` as a scalar input being used in a dynamic reshape or not.

Reply:
Wouldn't `-1` be allowed in the user program? If so, unless `-1` is supported, the frontend needs to analyze what `-1` should be replaced with, which isn't ideal. So, I think `-1` needs to be allowed.

Reply:
This case is failing because `kernel_runtime->getMaybeHeuristicsFor(args, forced_index_type)` is inserting the `-1` from `args` and then trying to get heuristics. This leaves the -1 in the output shape to the view op. `getConcretizationInfo` handles this problem by translating -1 appropriately to split/merge ops which have hte proper (positive) sizes. Since the output of the dynamic reshape will be discarded at concretization anyway, perhaps we just need a method that will update any negative extents of that `TensorView` with the concrete size.

Reply:
Alternatively, in the reshape op in `alias.cpp` we could try and build the actual expression for each extent, handling the -1 case with a `where`. But that's going to be a challenge to simplify later.

Reply:
Ah, so with `-1`, do you mean an `IterDomain` is concretized to have extent `-1`? It seems like this is an issue of concretization, not necessarily about the executor extension.